pkg/controller/dns/cluster_external_dns.go

```diff
@@ -0,0 +1,108 @@
+package dns
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/Azure/aks-app-routing-operator/api/v1alpha1"
+	"github.com/Azure/aks-app-routing-operator/pkg/config"
+	"github.com/Azure/aks-app-routing-operator/pkg/controller/controllername"
+	"github.com/Azure/aks-app-routing-operator/pkg/controller/metrics"
+	"github.com/Azure/aks-app-routing-operator/pkg/util"
+	"github.com/go-logr/logr"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var ClusterExternalDNSControllerName = controllername.New("cluster", "externaldns", "crd")
+
+func NewClusterExternalDNSController(mgr ctrl.Manager, config *config.Config) error {
+	return ClusterExternalDNSControllerName.AddToController(
+		ctrl.NewControllerManagedBy(mgr).
+			For(&v1alpha1.ClusterExternalDNS{}).
+			Owns(&appsv1.Deployment{}).
+			Owns(&corev1.ConfigMap{}), mgr.GetLogger()).Complete(
```

Comment:
We don't have to change this but I don't know if we want to watch Configmaps. 

This adds it to our cache and large clusters have many many configmaps. I think it's probably better to just not watch so we don't have memory issues on larger clusters. Definitely a tradeoff there though. If customers are touching our configmaps they break themselves but we will rereconcile eventually (after sync interval of 10 hours). Definitely a tradeoff

Reply:
Honestly with that said - I don't mind removing. Reconciliation should take effect to take over any temporary misconfigs, and if they do break themselves by messing w configmap it's not like they're breaking existing DNS records/breaking their apps - it's just temporarily halting syncing until we re-reconcile the correct configmap. So I can remove this